rbc/omniscidb.py

```diff
@@ -830,6 +830,9 @@ def is_udtf(sig):
         fid = 0  # UDF/UDTF id
         udfs, udtfs = [], []
         for device, target_info in self.targets.items():
+            # Set a global target_info instance
+            TargetInfo.set_instance(target_info)
+
```

Comment:
This construction begs for using the `with` statement. Let's make `TargetInfo` a context manager (define methods `__enter__`, `__exit__`) so that one can use:
```
with target_info:
    ...
```
instead of
```
TargetInfo.set_instance(target_info)
...
TargetInfo.clear_instance()
```